lms/djangoapps/courseware/date_summary.py

```diff
@@ -227,6 +228,14 @@ def description(self):
     def date(self):
         return self.course.certificate_available_date
 
+    @property
+    def has_certificate_modes(self):
+        return any([
+            mode.slug for mode in CourseMode.modes_for_course(
+                course_id=self.course.id, include_expired=True
+            ) if mode.slug != CourseMode.AUDIT
```

Comment:
Is checking `AUDIT` enough?  Do we need to worry about `HONOR` and `NO_ID_PROFESSIONAL_MODE` here?

Reply:
@iloveagent57 `audit` is the only no-certificate mode. certs are applicable on honor all other modes. 
Thoughts?

Reply:
OK.  I'm just curious - is that captured somewhere in the `course_modes` directory?

Reply:
@iloveagent57  I have got this understanding by looking at the `ecom` https://ecommerce.edx.org/courses/new/
Only for audit mode its stating "No certificate."

It must have been documented somewhere, but I am not sure.

Reply:
Here is code from CMS certificate interface. 
https://github.com/edx/edx-platform/blob/master/cms/djangoapps/contentstore/views/certificates.py#L381

It is also applying check on `audit` mode.